chainstate/src/interface/chainstate_interface_impl.rs

```diff
@@ -382,4 +382,10 @@ impl<S: BlockchainStorage> ChainstateInterface for ChainstateInterfaceImpl<S> {
         )?;
         Ok(())
     }
+
+    fn utxo(&self, outpoint: &OutPoint) -> Option<Utxo> {
```

Comment:
This should be fallible

Reply:
You mean something like `Result<Option<Utxo>, ChainstateError>`? Because none of the methods called is a `Result`